### DISCUSSION TITLE
Keychain Access on restarting Parse-Server-Swift

### DISCUSSION THREAD

**Discussion by jaysonng** [netreconlab/parse-server-swift#37] [Q&A]
Hi @cbaker6 

I was wondering if you've encountered this popup. It happens everytime I restart the server and there's 3 popups of this consecutively where I have to enter my password 3x.

![Screenshot 2023-06-21 at 6 47 12 PM](https://github.com/netreconlab/parse-server-swift/assets/440445/6c359ba5-2292-4885-ad77-155c225ee84f)

In the Keychain Access app, I've even tried allowing all apps access 

![Screenshot 2023-06-21 at 7 08 07 PM](https://github.com/netreconlab/parse-server-swift/assets/440445/29f20eff-d99a-41b7-9edf-bf2e0f792375)

Do you know a fix?

I've searched but nothing has worked for me. 

thanks,

**Comment by cbaker6** [comment]
I don’t usually see this, but I don’t typically run the server in macOS, I run it from the docker image in which Linux doesn’t have a Keychain. Parse-Swift automatically configures the Keychain on Apple OS’s, which is probably why you are seeing this (similar to testing ParseSwift on macOS).

**Comment by jaysonng** [comment]
> Parse-Swift automatically configures the Keychain on Apple OS’s, which is probably why you are seeing this (similar to testing ParseSwift on macOS).

ah I see. I'll have to look into that then. thanks! :)
